http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java

```diff
@@ -2143,8 +2143,18 @@ private ByteBuf encodeBodyAsByteBuf(Object body, MediaTypeCodec codec, ChannelHa
                 return (Publisher<MutableHttpResponse<?>>) httpFilter.doFilter(requestReference.getAndSet(request), this);
             }
         };
-        HttpFilter httpFilter = filters.get(0);
-        return (Publisher<MutableHttpResponse<?>>) httpFilter.doFilter(requestReference.get(), filterChain);
+        Optional<HttpRequest<Object>> prevRequest = ServerRequestContext.currentRequest();
+        try {
+            ServerRequestContext.set(requestReference.get());
+            HttpFilter httpFilter = filters.get(0);
+            return (Publisher<MutableHttpResponse<?>>) httpFilter.doFilter(requestReference.get(), filterChain);
+        } finally {
+            if (prevRequest.isPresent()) {
+                ServerRequestContext.set(prevRequest.get());
+            } else {
+                ServerRequestContext.set(null);
+            }
+        }
```

Comment:
Could you not use `ServerRequestContext.with(requestReference.get(), () -> )`?

Reply:
I can but I don't know if it's worse to have a new lambda always being created there?